src/win/tty.c

```diff
@@ -3079,6 +3111,7 @@ static void CALLBACK uv__tty_console_resize_event(HWINEVENTHOOK hWinEventHook,
     uv_sem_post(&uv_tty_output_lock);
     uv__signal_dispatch(SIGWINCH);
     return;
+  } else {
+    uv_sem_post(&uv_tty_output_lock);
   }
-  uv_sem_post(&uv_tty_output_lock);
 }
```

Comment:
>libuv@7d950c0 uses a mutex here additionally: libuv@7d950c0#diff-16746be892849b2c5daf1be3613ed28cR2377-R2384
> 
>This might be the reason for getting too many (flaky) SIGWINCH events in Neovim's tests now.

Since `uv__tty_console_output_handle` is used in the next place https://github.com/erw7/libuv/blob/libuv-v1.32.0/src/win/tty.c#L738
 in another thread, I decided that it needs to be locked using `uv_tty_output_lock`.

I experimentally created a branch https://github.com/erw7/libuv/tree/libuv-v1.32.0-test
 that reverted to the mutex like the original change. Can you try this once?

> Do functionaltests pass for you?

When I try it locally, I don't have any problems with `SIGWINCH`.